### PR TITLE
Dragonberry

### PR DESCRIPTION
**Pull request by asalzmann**
## Context and purpose of the change

An IBC security advisory was issued ([Dragonberry](https://forum.cosmos.network/t/ibc-security-advisory-dragonberry/7702)), this PR applies the updates recommended by the advisory. 

## Brief Changelog

- Add replace in go.mod
- Bump SDK to 0.45.9

## Author's Checklist

I have...

- [ ] Run and PASSED locally all GAIA integration tests
- [ ] If the change is contentful, I either:
    - [ ] Added a new unit test OR 
    - [ ] Added test cases to existing unit tests
- [ ] OR this change is a trivial rework / code cleanup without any test coverage

If skipped any of the tests above, explain.

## Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] manually tested (if applicable)
- [ ] confirmed the author wrote unit tests for new logic
- [ ] reviewed documentation exists and is accurate

## Documentation and Release Note

  - [ ] Does this pull request introduce a new feature or user-facing behavior changes? 
  - [ ] Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`?
  - [ ] This pull request updates existing proto field values (and require a backend and frontend migration)? 
  - [ ] Does this pull request change existing proto field names (and require a frontend migration)?
  How is the feature or change documented? 
      - [ ] not applicable
      - [ ] jira ticket `XXX` 
      - [ ] specification (`x/<module>/spec/`) 
      - [ ] README.md 
      - [ ] not documented

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -134,16 +134,14 @@ replace (
...
-	// logic and push a new tag and the branch to github (use that tag below)
+	// apply ICA patch
+	github.com/cosmos/cosmos-sdk => github.com/Stride-Labs/cosmos-sdk v0.45.9-dragonberry
 	github.com/gogo/protobuf => github.com/regen-network/protobuf v1.3.3-alpha.regen.1
 	// Use a specific version of ignite cli
 	github.com/ignite-hq/cli => github.com/ignite-hq/cli v0.21.0
 	// Use the go-keychain
 	github.com/keybase/go-keychain => github.com/99designs/go-keychain v0.0.0-20191008050251-8e49817e8af4
-	// fork cast, not really sure why
+	// fork cast to add additional error checking
```
Comment (faddat) [inline go.mod:144]: 🔥

**Review by faddat — APPROVED** [review]
When I handled that test, I did it a little differently-- check out notional-labs/stride in the dragonberry branch.

Otherwise I think our changes are 1:1 the same.

File: go.mod [hunk 2]
```diff
@@ -79,69 +76,73 @@ require (
...
-	// of the target version on github.com/Stride-Labs/cosmos-sdk, then remove the error redaction
-	// logic and push a new tag and the branch to github (use that tag below)
-	github.com/cosmos/cosmos-sdk => github.com/Stride-Labs/cosmos-sdk v0.45.4-debug-2
+	// Use the keyring specified by the cosmos-sdk
+	github.com/99designs/keyring => github.com/cosmos/keyring v1.1.7-0.20210622111912-ef00f8ac3d76
+
+	// dragonberry replkace line per: https://github.com/cosmos/cosmos-sdk/releases/tag/v0.45.9
+	github.com/confio/ics23/go => github.com/cosmos/cosmos-sdk/ics23/go v0.8.0
+	// apply ICA patch
+	github.com/cosmos/cosmos-sdk => github.com/Stride-Labs/cosmos-sdk v0.45.9-dragonberry
```
Comment (asalzmann) [inline go.mod:138]: @faddat added a replace here to capture https://github.com/Stride-Labs/cosmos-sdk/commit/35d2308b9206e462765ee7e2c902cb5736396a8f

Comment (faddat) [reply go.mod:138]: so I'd ended up pushing to the wrong place-- few small differences between our work, but think that both are safe:
[
](https://github.com/notional-labs/stride/tree/dragonberry)

**Pull request merged**